### ISSUE TITLE
Show player crosshair when spectating and record in demo

### ISSUE THREAD

**Issue by TsFreddie** [teeworlds/teeworlds#2711]
There are many attempts of various anticheats being developed as mods. But sometime it is plenty enough to spectate players' crosshair.

Also it is just a nice to have feature for demos since you can record the mouse movement when you played it.

A toggle and simple interpolation between tick is needed as well of course. Crosshair interpolation may not reproduce the mouse movement exactly, but spectating crosshair should looks smooth at least by default.

**Comment by Dune-jr** [comment]
Sounds good to me, I see no downside to that

**Comment by oy** [comment]
Showing it when playing a demo shouldn't be a problem. Doing so when spectating someone needs a network change as clients only get the aim angle of a player and not the target pos.

**Comment by TsFreddie** [comment]
Indeed. I don't know if this should be in 0.8 roadmap or have a temporary NetObject like the race info.

Although I would advise against put target pos into CharacterCore:
1. Having cursor info in all character could enable more cheats like anti-aim or auto-kill (or made them easier to achieve).
2. We only need one players cursor info, and only while spectating anyway.

**Comment by oy** [comment]
Could put it in the spectator info.
